Doc/library/argparse.rst

```diff
@@ -26,6 +26,61 @@ module also automatically generates help and usage messages and issues errors
 when users give the program invalid arguments.
 
 
+Summary
+-------
+
+Core function
+^^^^^^^^^^^^^
+
+.. csv-table::
+    :header: "Name", "Common Used Arguments", "Example"
+
+    ":class:`argparse.ArgumentParser`", "`prog`_, `description`_, `formatter_class`_", "``parser = argparse.ArgumentParser(prog='PROG', description='DESC', formatter_class=argparse.RawDescriptionHelpFormatter)``"
+    ":func:`ArgumentParser.add_argument`", "`name or flags`_, `action`_, `default`_, `type`_, `required`_, `help`_", "``parser.add_argument('-v', '--verbose', action='store_true', help='Show various debugging information')``"
+
+
+Basic Usage of :func:`add_argument`
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+
+**Name or Flags Type**
+
+====================== ===========================
+Type                   Example
+====================== ===========================
+Positional             ``'foo'``
+Optional               ``'-v'``, ``'--verbose'``
+====================== ===========================
+
+
+**Basic Arguemnts:**
+
+====================== =============================================== =========================================================================================================================
+Name                   Description                                     Keywords
+====================== =============================================== =========================================================================================================================
+`action`_              What to do on arguments                         ``'store'``, ``'store_const'``, ``'store_true'``, ``'append'``, ``append_const``, ``count``, ``help``, ``version``
+`default`_             Default value of arguments
+`type`_                Treat arguments as type                         ``int``, ``float``, ``bool``, ``argparse.FileType('w')``, ``callable function``
+`help`_                Help message of arguments
+====================== =============================================== =========================================================================================================================
+
+
+
+**Advance Arguments:**
+
+====================== ================================================== =======================================================================================================================
+Name                   Description                                        Keywords
+====================== ================================================== =======================================================================================================================
+`nargs`_               Associates number of arguments                     ``N`` (an integer), ``'?'``, ``'*'``, ``'+'``, ``argparse.REMAINDER``
+`const`_               The constant values of name or flags
+`choices`_             Arguments should selected from this set or values  ``['foo', 'bar']``, ``range(1, 10)``, Any object that support ``in`` operator
```

Comment:
"A container that lists the possible values"

(container could link to the glossary entry if there is one)